### ISSUE TITLE
`argocd version` and other commands result in error when not in argocd namespace, even with `--namespace` option

### ISSUE THREAD

**Issue by BalzGuenat** [argoproj/argo-cd#11348]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

While not in the `argocd` namespace, the command `argocd version --namespace argocd` fails with the error `FATA[0001] configmap "argocd-cm" not found`. This started to happen after updating the Argo CD CLI. We're using the core install.

**To Reproduce**

1. Set the namespace of your current K8s context to something other than `argocd` (i.e. the NS where Argo CD is installed).
2. Run `argocd version --namespace argocd`

**Expected behavior**

No error occurs.

**Screenshots**

**Version**

```shell
argocd: v2.5.2+148d8da.dirty
  BuildDate: 2022-11-07T18:41:19Z
  GitCommit: 148d8da7a996f6c9f4d102fdd8e688c2ff3fd8c7
  GitTreeState: dirty
  GoVersion: go1.19.3
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.5.2+148d8da.dirty
  BuildDate: 2022-11-07T18:41:19Z
  GitCommit: 148d8da7a996f6c9f4d102fdd8e688c2ff3fd8c7
  GitTreeState: dirty
  GoVersion: go1.19.3
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v4.5.5 2022-05-20T20:21:22Z
  Helm Version: v3.9.3+g414ff28
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.18.0
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jarviliam** [comment]
Yeah seems to be an issue with the core installation. It seems that the headless server doesn't use the --namespace flag and defaults to the namespace in the user's kube context. 

@crenshaw-dev I can submit fix for this if you don't mind assigning me to it 👍 .
